packages/tokens/src/rwa/compliance/storage.rs

```diff
@@ -201,16 +201,17 @@ pub fn remove_module_from(e: &Env, hook: ComplianceHook, module: Address) {
 /// * `from` - The address that sent the tokens.
 /// * `to` - The address that received the tokens.
 /// * `amount` - The amount of tokens transferred.
+/// * `contract` - The address of the contract that is performing the transfer.
 ///
 /// # Cross-Contract Calls
 ///
 /// Invokes `on_transfer(from, to, amount)` on each registered module.
-pub fn transferred(e: &Env, from: Address, to: Address, amount: i128) {
+pub fn transferred(e: &Env, from: Address, to: Address, amount: i128, contract: Address) {
     let modules = get_modules_for_hook(e, ComplianceHook::Transferred);
 
     for module_address in modules.iter() {
         let client = ComplianceModuleClient::new(e, &module_address);
-        client.on_transfer(&from, &to, &amount);
+        client.on_transfer(&from, &to, &amount, &contract);
     }
 }
```

Comment:
```suggestion
pub fn transferred(e: &Env, from: Address, to: Address, amount: i128, token: Address) {
  token.require_auth();
  // also check token is bound
  let modules = get_modules_for_hook(e, ComplianceHook::Transferred);

  for module_address in modules.iter() {
    let client = ComplianceModuleClient::new(e, &module_address);
        client.on_transfer(&from, &to, &amount, &token);
  }
}
```

I think we have to enforce authorization here, because this should be callable only by bound tokens.
I also suggest we rename `contract` to `token`, because this is not just some kind of a contract, but a token specifically.

Same comment goes for all the other functions below.

Reply:
makes sense, will address the suggestions 👍